src/utils/getTokenList.ts

```diff
@@ -1,11 +1,14 @@
-import { TokenList } from '@uniswap/token-lists'
+import { ChainId } from '@kyberswap/ks-sdk-core'
 import axios from 'axios'
 
+import { TokenMap } from 'hooks/Tokens'
+import { TokenInfo, WrappedTokenInfo } from 'state/lists/wrappedTokenInfo'
+import { isAddress } from 'utils'
+
 import { getFormattedAddress } from './tokenInfo'
 
-// lazily get the validator the first time it is used
 // loop to fetch all whitelist token
-export async function getTokenListV2(listUrl: string): Promise<TokenList> {
+export async function getTokenList(listUrl: string, chainId: ChainId): Promise<TokenMap> {
   return new Promise(async (resolve, reject) => {
```

Comment:
Using async inside Promise constructor is like using Promise inside Promise. I think it's Promise is not needed, async/await is enough here. Maybe remove new Promise?

Reply:
yes, I remember I removed it before but it's still here 🐯 =))